Source/Core/InputCommon/ControlReference/ExpressionParser.cpp

```diff
@@ -21,6 +24,55 @@ namespace ciface::ExpressionParser
 {
 using namespace ciface::Core;
 
+class ControlExpression;
+
+class HotkeySuppressions
+{
+public:
+  using Modifiers = std::vector<std::unique_ptr<ControlExpression>>;
+
+  struct InvokingDeleter
+  {
+    template <typename T>
+    void operator()(T* func)
+    {
+      (*func)();
+    }
+  };
+
+  using Suppressor = std::unique_ptr<std::function<void()>, InvokingDeleter>;
+
+  bool IsSuppressed(Device::Input* input) const
+  {
+    // Input is suppressed if it exists in the map at all.
+    return m_suppressions.lower_bound({input, nullptr}) !=
```

Comment:
I feel like I should know this but what are you doing here exactly?  How are you lower-bounding the pointer and why are you comparing against the lower-bound of the pair of pointers with another +1?

Reply:
They're just sorted based on the pointer values. The `lower_bound` trick is to test if any pair in the map has `input` for the `first` value.